packages/vite-node/src/client.ts

```diff
@@ -56,9 +56,11 @@ export class ViteNodeRunner {
       // probably means it was passed as variable
       // and wasn't transformed by Vite
       if (this.shouldResolveId(dep)) {
+        if (extname(dep)) dep = normalize(`${dirname(id)}/${dep}`)
```

Comment:
So after digging up I know why I didn't like this code from the first glance.

`id` is treated as `/src/path` instead of `${root}/src/path` - this is why `resolveId` doesn't work. we should combine `root` and `id` when needed and pass it to `this.options.resolveId`. or better teach `resolveId` how to handle imports without root in them

Reply:
ID looked like the path from the current directory to the file that has the import.  I needed to check extname otherwise I was hitting problems with node internals and package names, but I guess shouldResolveId is handling those cases now?

Regardless I was just trying fix one of the few problems remaining so I could fully adopt this library for my companies code base, any suggested fix for my problem will be fine for me since you know the code base better. I like trying to provide a solution rather then making another issue.  I can add the above to the server.ts code in my PR.

Reply:
Try to add this to `server.ts` in your PR, yeah. The `dep` value is fine - since it is not absolute, it will be resolved according to `importer`, but `importer` was incorrect. So the solution to your problem would be to fix `importer`.

You also can import files without specifying the extension name, so the problem can rise again. I think the suggested solution is more well around.

Reply:
I updated my PR, I think I added it in the right spot